### PR TITLE
gracefully fall back to non pid locked precompilation if FileWatching is not present in sysimage

### LINKED ISSUE #56569: FileWatching required in sysimage for parellel precompilation to work otherwise ungracefully errors.
This line

https://github.com/JuliaLang/julia/blob/55bdb54dd2e1b0ea6df017f74f9af1de7abb8038/base/precompilation.jl#L1036

fails if one builds a sysimage without FileWatching. It should instead probably run without the pid locking if FileWatching does not exist in the sysimage, or try load it. I tried to change it to using `require_stdlib` but this code gets run during Pkg precompilation and then we get an error trying to use `require_stdlib` during package precompilation.

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes #56569

### CODE REVIEW

**Review by vtjnash — COMMENTED** [review]
This is really not the correct way to hijack state from another library by hard coding its name. We have hooks for this in Base already:
```
   3819 # Set by FileWatching.__init__()
   3820 global mkpidlock_hook::Any
   3821 global trymkpidlock_hook::Any
   3822 global parse_pidfile_hook::Any
```

**Comment by KristofferC** [comment]
Better?

File: base/precompilation.jl [hunk 1]
```diff
@@ -1031,14 +1031,17 @@ end
 
 # Can be merged with `maybe_cachefile_lock` in loading?
 function precompile_pkgs_maybe_cachefile_lock(f, io::IO, print_lock::ReentrantLock, fancyprint::Bool, pkg_config, pkgspidlocked, hascolor)
+    FileWatching = get(Base.loaded_modules, Base.PkgId(Base.UUID("7b1f6079-737a-58dc-b8bc-7a2ca5c1b5ee"), "FileWatching"), nothing)
```
Comment (vtjnash) [inline base/precompilation.jl:1034]:
```suggestion
```

File: base/Base_compiler.jl [hunk 2]
```diff
@@ -266,6 +266,21 @@ function strcat(x::String, y::String)
     return out
 end
 
+function file_exists(filename::String)
```
Comment (vtjnash) [inline base/Base_compiler.jl:269]: This seems unrelated? Normally you'd call `access` with `F_OK` to implement this

Comment (KristofferC) [reply base/Base_compiler.jl:269]: Eh, I put some commits on this branch by accident it seems..

Comment (KristofferC) [reply base/Base_compiler.jl]: Fixed

**Pull request merged**